### PR TITLE
first version for the scoring system (numerical)

### PR DESCRIPTION
**Pull request by trevorgh**

### CODE REVIEW

**Review by dylan-mankin — APPROVED** [review]

File: backend/backend/scoring.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import typing
+
+
+# these are the rotios we are gonna use and their weights(0-1)
```
Comment (mileswatson) [inline backend/backend/scoring.py:4]: ratios

File: backend/backend/scoring.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ratios_weights = {
+    "current_ratio": 0.5,
+    "roic": 0.9,
+    "debt_to_equity": 0.7
+    }
+
+max_score = 0
```
Comment (mileswatson) [inline backend/backend/scoring.py:11]: sum(dict.values()) maybe?

File: backend/backend/scoring.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+for key in ratios_weights:
+    max_score += ratios_weights[key]
+
+
+
+def evaluate_current_ratio(value: float) -> float:
```
Comment (mileswatson) [inline backend/backend/scoring.py:18]: Remove duplicate code with a helper function.

File: backend/backend/scoring.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    # calculate ratios
+    ratios["current_ratio"] = current_assets / current_liabilities
+    ratios["roic"] = (net_income - dividends) / (debt + equity)
+    ratios["debt_to_equity"] = total_liabilities / equity
+
+    # use the scoring system
+    print(scoring(ratios))
+
+
+
```
Comment (mileswatson) [inline backend/backend/scoring.py:121]: Remove extra lines - two lines max.

File: backend/backend/scoring.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    for key in scoring_dic:
+        if key > value:
+            score = scoring_dic[key]
+            break
+    
+    return score
+
+
+
+def scoring(ratios: typing.Dict[str, float]) -> float:
```
Comment (mileswatson) [inline backend/backend/scoring.py:83]: `from typing import Dict` at start of file
Maybe create a class to represent the ratios? Currently difficult to know what each key should be.

**Review by mileswatson — CHANGES_REQUESTED** [review]

**Review by mileswatson — APPROVED** [review]

**Pull request merged**